pkg/sync/expand.go

```diff
@@ -112,19 +190,19 @@ func isInCycle(newCycle []string, cycles [][]string) bool {
 	return false
 }
 
-func shift(arr []string, n int) []string {
+func shift(arr []int, n int) []int {
 	for i := 0; i < n; i++ {
 		arr = append(arr[1:], arr[0])
 	}
 	return arr
 }
 
-func (d *EntitlementGraph) getCycle(visits []string) ([]string, bool) {
-	entitlementID := visits[len(visits)-1]
-	for descendantID := range d.GetDescendants(entitlementID) {
-		tempVisits := make([]string, len(visits))
+func (g *EntitlementGraph) getCycle(visits []int) ([]int, bool) {
+	nodeId := visits[len(visits)-1]
```

Comment:
do we need to assert that `len(visits) != 0` here?

Reply:
I can add it, but getCycle() is only called in one place, and it always has one element in that array argument.